server/keyshare/tasks/tasks.go

```diff
@@ -128,16 +128,20 @@ func (t *taskHandler) sendExpiryEmails(ctx context.Context, id int64, username,
 			}
 
 			if err := keyshare.VerifyMXRecord(email); err != nil {
-				if t.revalidateMail {
-					if err = t.db.ExecUserContext(ctx, "UPDATE irma.emails SET revalidate_on = $1 WHERE id = $2",
-						time.Now().AddDate(0, 0, 5).Unix(),
-						id); err != nil {
-						t.conf.Logger.WithField("error", err).Error("Could not update email address to set revalidate_on")
-						return err
-					}
+				if !t.revalidateMail {
+					// It could be this user has another, valid email address. Therefore we do not abort
+					// since is a chance we can still inform the user.
+					return nil
 				}
 
-				// We wait with further processing until the email address is revalidated
+				if revErr := t.db.ExecUserContext(ctx, "UPDATE irma.emails SET revalidate_on = $1 WHERE id = $2",
+					time.Now().AddDate(0, 0, 5).Unix(),
+					id); revErr != nil {
+					t.conf.Logger.WithField("error", revErr).Error("Could not update email address to set revalidate_on")
+					return revErr
+				}
+
+				// Abort to prevent sending mail until the invalid email address is revalidated
 				// so we can send the expiry mail to all, and only valid, addresses at once
 				return err
```

Comment:
The problem here was that `err = t.db.ExecUserContext(ctx, "UPDATE irma.emails SET revalidate_on = $1 WHERE id = $2"` overrides the existing error with `nil`. This means that this function returned nil instead of an error if the revalidation was executed.

Reply:
By using `revErr,` `err` is not overwritten with `nil` so it does return `keyshare.ErrInvalidEmail` and/or `keyshare.ErrInvalidEmailDomain` and any other type of exception to `expireAccounts(..)`. I agree this it not the most beautiful pattern, but did not see another way to do it without a much bigger refactoring.